### PR TITLE
Project Happy Thoughts API - Jessika Hage

### PR DESCRIPTION
**Pull request by jessika-hage**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,30 +1,164 @@
...
+  const { page, sort } = req.query;
+  const pageSize = 20;
+
+  // Skip 20 results per page
+  const pageResults = (page) => {
+    return ((page - 1) * pageSize)
+  };
+
+  // Sort thoughts on either most liked, newest or oldest.
+  // Default set to newest
```
Comment (sssylvasss) [inline server.js:63]: Nice with comments!

**Review by sssylvasss — APPROVED** [review]
I have nothing to add! Everything is there and it works great! Very happy happy thoughts:)

File: server.js [hunk 2]
```diff
@@ -1,30 +1,164 @@
...
+  const sortThoughts = (sort) => {
+    if (sort === "hearts") {
+      return { hearts: -1 };
+    } else if (sort === "oldest") {
+      return { createdAt: 1 };
+    } else {
+      return { createdAt: -1 };
+    }
+  };
```
Comment (KarolCislowski) [inline server.js:72]: Nicely provided sorting and pagination 😻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,164 @@
...
+    res.status(400).json({ message: 'Something went wrong', error });
+  }
+});
+
+// POST request for sending thought
+app.post('/thoughts', async (req, res) => {
+  const { message, name } = req.body;
+
+  try {
+    const newThought = await new Thought({ message, name: name || 'Anonymous' }).save();
```
Comment (KarolCislowski) [inline server.js:97]: Great way to provide a default value!

File: server.js [hunk 4]
```diff
@@ -1,30 +1,164 @@
...
+});
+
+// POST request for likes
+app.post('/thoughts/:id/like', async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate(
+      id,
+      { $inc: { hearts: 1 } },
```
Comment (KarolCislowski) [inline server.js:111]: 👍🏻

File: server.js [hunk 5]
```diff
@@ -1,30 +1,164 @@
...
+    const deletedThought = await Thought.findByIdAndDelete(id);
+    if (deletedThought) {
+      res.json(deletedThought)
+    } else {
+      res.status(404).json({ message: 'Not found!' })
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error });
+  }
+});
```
Comment (KarolCislowski) [inline server.js:158]: Nice endpoints 👍🏻 I'm waiting for adding these functionalities to the frontend. But remember to save it only to a post owner, so you need to provide some auth here 🤓

**Review by KarolCislowski — APPROVED** [review]
Great API 🦄

**Pull request closed without merging**